### PR TITLE
Add test for benchmark-test jenkinsfile job

### LINKED ISSUE #3541: Add unit test for benchmark test jenkinsfile
Add unit-tests for `benchmark-test.jenkinsfile`.

### PR DESCRIPTION
**Pull request by rishabh6788**
### Description
Add test for benchmark-test jenkinsfile job. 

### Issues Resolved
#3541 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: tests/jenkins/TestRunNonSecBenchmarkTestScript.groovy [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        binding.setVariable('WEBHOOK_URL', 'test://artifact.url')
+
+        super.setUp()
+    }
+
+    @Test
+    public void testRunNonSecurityBenchmarkTestScript_verifyPipeline() {
+        super.testPipeline("jenkins/opensearch/benchmark-test.jenkinsfile",
+                "tests/jenkins/jenkinsjob-regression-files/opensearch/benchmark-test.jenkinsfile")
+    }
```
Comment (gaiksaya) [inline tests/jenkins/TestRunNonSecBenchmarkTestScript.groovy:97]: Looks like we are adding just 1 txt file where security is set to `false` I believe we can add one more?

```
    @Test
    public void testRunSecurityBenchmarkTestScript_verifyPipeline() {
        super.testPipeline("jenkins/opensearch/benchmark-test.jenkinsfile",
                "tests/jenkins/jenkinsjob-regression-files/opensearch/benchmark-test-without-security.jenkinsfile")
    }

File: tests/jenkins/TestRunNonSecBenchmarkTestScript.groovy [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            shCommand -> shCommand.contains('bucket')
+        }
+
+        assertThat(s3DownloadCommands.size(), equalTo(2))
+        assertThat(s3DownloadCommands, hasItem(
+                "{file=config.yml, bucket=ARTIFACT_BUCKET_NAME, path=test_config/config.yml, force=true}".toString()
+        ))
+        assertThat(s3DownloadCommands, hasItem(
+                "{file=benchmark.ini, bucket=ARTIFACT_BUCKET_NAME, path=test_config/benchmark.ini, force=true}".toString()
+        ))
```
[the remark this answers is not in the document]
Comment (rishabh6788) [reply tests/jenkins/TestRunNonSecBenchmarkTestScript.groovy]: Fixed!

File: tests/jenkins/TestRunNonSecBenchmarkTestScript.groovy [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        binding.setVariable('HAS_SECURITY', true)
+        helper.registerAllowedMethod("cfnDescribe", [Map.class]) { throw exception }
+        helper.registerAllowedMethod('sh', [String.class], { String cmd ->
+            updateBuildStatus('ABORTED')
+        })
+        runScript("jenkins/opensearch/benchmark-test.jenkinsfile")
+
+        assertJobStatusAborted()
+        assertCallStack()
+        printCallStack()
```
[the remark this answers is not in the document]
Comment (rishabh6788) [reply tests/jenkins/TestRunNonSecBenchmarkTestScript.groovy]: Removed!

File: tests/jenkins/TestRunNonSecBenchmarkTestScript.groovy (hunk 1 above)
Comment (rishabh6788) [reply tests/jenkins/TestRunNonSecBenchmarkTestScript.groovy:98]: Added tests for Security as well.

File: tests/jenkins/TestRunNonSecBenchmarkTestScript.groovy [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        ))
+
+        def s3DownloadCommands = getCommandExecutions('s3Download', 'bucket').findAll {
+            shCommand -> shCommand.contains('bucket')
+        }
+
+        assertThat(s3DownloadCommands.size(), equalTo(2))
+        assertThat(s3DownloadCommands, hasItems(
+                "{file=config.yml, bucket=ARTIFACT_BUCKET_NAME, path=test_config/config.yml, force=true}".toString(),
+                "{file=benchmark.ini, bucket=ARTIFACT_BUCKET_NAME, path=test_config/benchmark.ini, force=true}".toString()
```
Comment (gaiksaya) [inline tests/jenkins/TestRunNonSecBenchmarkTestScript.groovy:120]: nit:
```suggestion
                "{file=config.yml, bucket=ARTIFACT_BUCKET_NAME, path=test_config/config.yml, force=true}",
                "{file=benchmark.ini, bucket=ARTIFACT_BUCKET_NAME, path=test_config/benchmark.ini, force=true}".
```

**Review by gaiksaya — APPROVED** [review]
LGTM!

**Pull request merged**